tiny-event-sourcing-lib/src/main/kotlin/ru/quipy/saga/SagaInfo.kt

```diff
@@ -0,0 +1,85 @@
+package ru.quipy.saga
+
+import java.util.*
+
+/**
+ * The SagaContext class represents the context of a Saga, containing relevant information about the Saga's execution.
+ * It is used to track the state and progression of the Saga.
+ *
+ * The [ctx] field is an extended version for working with Sagas. It is updated using the [SagaManager].
+ * The [correlationId], [currentEventId], and [causationId] fields provide a simplified version for working with Sagas
+ * and are always used.
+ *
+ * @property ctx A map containing information about the Saga,
+ * where the key is the Saga name and the value is the corresponding [SagaInfo] instance.
+ * @property correlationId The correlation ID for all events in the Saga. It is equivalent to [SagaInfo.sagaInstanceId].
+ * @property currentEventId The ID of the current step in the Saga.
+ * It is equivalent to [SagaInfo.sagaStepId].
+ * @property causationId The causation ID, representing the previous step in the Saga.
+ * It is equivalent to [SagaInfo.prevStepsIds].
+ */
+data class SagaContext(
+    val ctx: Map<String, SagaInfo> = mapOf(),
+    var correlationId: UUID? = null,
+    var currentEventId: UUID? = null,
+    var causationId: UUID? = null
+)
+
+/**
+ * The SagaInfo class represents information about a step in a Saga.
+ * @property sagaInstanceId The ID of the Saga instance to which this step belongs.
+ * @property stepName The name of the step in the Saga.
+ * @property sagaStepId The ID that uniquely identifies this step in the Saga.
+ * @property prevStepsIds The set of IDs representing the previous steps in the Saga.
+ * @property stepIdPrevStepsIdsAssociation A map associating step IDs with their corresponding previous step IDs.
+ * Used when processing parallel steps in [SagaContext.plus].
+ */
+data class SagaInfo(
+    val sagaInstanceId: UUID,
+    val stepName: String,
+    val sagaStepId: UUID,
+    val prevStepsIds: Set<UUID> = setOf(),
+    val stepIdPrevStepsIdsAssociation: Map<UUID, Set<UUID>> = mapOf()
```

Comment:
why we need `stepIdPrevStepsIdsAssociation` if we have `prevStepsIds `?

Reply:
In case of parallel Saga steps, `stepIdPrevStepsIdsAssociation` is needed to know exactly that these previous steps should be used for the current step. Because in case we used only `prevStepsIds`, it will be unclear what information is contained there - previous steps for the previous step (this information is of no use when processing the current step), or it already contains previous steps for **THIS** step - the "+" operator for contexts was used.
Probably it is enough to do something like `currentPrevStepsIds: Set<UUID>` here